### PR TITLE
fix: LBP rounding

### PR DESCRIPTION
**Pull request by Roznovjak**
### Description
This PR rewrites the LBP math to use `multiply_by_rational_with_rounding` and
a different fixed point type (`U32F96`) that is more appropriate for the numbers
encountered in the math. This allows us to reduce rounding errors and so apply
fewer extra additions.

This PR also adds LBP invariant tests.

### CODE REVIEW

File: math/src/transcendental.rs [hunk 1]
```diff
@@ -428,12 +429,12 @@ mod tests {
 
 		assert_eq!(
 			pow(S::from_num(22.1234), S::from_num(2.1)),
-			Ok(D::from_num(667.096912176457))
+			Ok(D::from_str("667.0969121771803182631954923946").unwrap())
```
Comment (enthusiastmartin) [inline math/src/transcendental.rs:432]: wolfram result : 667.09691217718012362888000511134534475469139572779466865919

File: math/src/transcendental.rs [hunk 2]
```diff
@@ -428,12 +429,12 @@ mod tests {
...
 		);
 
 		assert_eq!(
 			pow(S::from_num(0.986069911074), S::from_num(1.541748732743)),
-			Ok(D::from_num(0.978604514488))
+			Ok(D::from_str("0.97860451447489653592682845716").unwrap())
```
Comment (enthusiastmartin) [inline math/src/transcendental.rs:437]: wolfram result: 
0.9786045144748965444712335043101175220576942351573728942167

File: pallets/lbp/src/lib.rs [hunk 3]
```diff
@@ -800,11 +809,15 @@ impl<T: Config> Pallet<T> {
...
 		ensure!(
 			!pool_data.initial_weight.is_zero()
 				&& pool_data.initial_weight < MAX_WEIGHT
+				&& pool_data.initial_weight >= MAX_WEIGHT / 50	// 2%
 				&& !pool_data.final_weight.is_zero()
-				&& pool_data.final_weight < MAX_WEIGHT,
+				&& pool_data.final_weight < MAX_WEIGHT
+				// when initial and final weights are >= 2%, then the weights are also <= 98%
+				&& pool_data.final_weight >= MAX_WEIGHT / 50, // 2%,
 			// TODO people could leak value out the pool if initial weight is < final weight due to fee structure
```
Comment (enthusiastmartin) [inline pallets/lbp/src/lib.rs:821]: what this TODO mean ?

File: math/src/transcendental.rs (hunk 1 above)
Comment (mrq1911) [reply math/src/transcendental.rs:432]: looks to me that is more accurate than before

**Review by mrq1911 — APPROVED** [review]

File: pallets/lbp/src/lib.rs (hunk 3 above)
Comment (Roznovjak) [reply pallets/lbp/src/lib.rs:821]: Hmm. I don't know. I didn't add this comment.

**Pull request merged**